internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -355,6 +357,88 @@ func resourceComputeClusterRead(d *pluginsdk.ResourceData, meta interface{}) err
 	return tags.FlattenAndSet(d, computeResource.Model.Tags)
 }
 
+func resourceComputeClusterUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.Workspaces
+	client := meta.(*clients.Client).MachineLearning.MachineLearningComputes
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	workspaceID, err := workspaces.ParseWorkspaceID(d.Get("machine_learning_workspace_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id, err := machinelearningcomputes.ParseComputeID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	workspace, err := mlWorkspacesClient.Get(ctx, *workspaceID)
```

Comment:
Why do we need to get the workspace here? Can't we retrieve all the info on the Compute Cluster by calling `client.ComputeGet`?

Reply:
The [SKU](https://learn.microsoft.com/en-us/rest/api/azureml/compute/create-or-update?view=rest-azureml-2024-04-01&tabs=HTTP#request-body) used by the compute cluster resources is the `SKU` of the workspace. And compute GET API will not return the SKU, but nil.

Reply:
That sounds like a bug we should raise on the Rest API Spec, can you please open one?